UniversityTimetablingSystem/backend/university_timetabling/timetable/views.py

```diff
@@ -1,156 +1,343 @@
-    except subprocess.CalledProcessError as e:
-        print(f"Error occurred: {e}")
-        return HttpResponseBadRequest(f"Error occurred: {e}")
     except Exception as e:
         print(f"Error occurred: {e}")
-        return HttpResponseBadRequest(f"Error occurred: {e}")
+        return HttpResponseServerError(str(e))
+
+
+def process_availability(availability_data):
+    availability = []
+    for slot in availability_data:
+        time_slot = {
+            University.DAY: slot[University.DAY],
+            University.START_TIME: slot[University.START_TIME],
+            University.END_TIME: slot[University.END_TIME]
+        }
+        if not all(time_slot.values()):
+            return HttpResponseBadRequest('Missing fields in availability slot')
+        availability.append(time_slot)
+    return availability
+
+
+def process_preferred_courses(preferred_courses_data):
+    preferred_courses = []
+    for course_data in preferred_courses_data:
+        if University.COURSE_NAME not in course_data:
+            return HttpResponseBadRequest('Missing courseName in preferredCourses')
+
+        course_name = course_data[University.COURSE_NAME]
+        if not course_name:
+            return HttpResponseBadRequest('Course name cannot be empty')
+
+        print(f"Processing course: {course_name}")
+
+        try:
+            course_doc = db.collection(University.COURSES).get(course_name)
+            print(course_doc)
+            if not course_doc:
+                return HttpResponseBadRequest(f"Course {course_name} does not exist in the database.")
+        except Exception as e:
+            return HttpResponseBadRequest(f"Error retrieving course {course_name}: {str(e)}")
+
+        try:
+            time_slots = course_doc.get(University.PREFERRED_TIME_SLOTS, [])
+            preferred_time_slots = [{
+                University.DAY: slot[University.DAY],
+                University.START_TIME: slot[University.START_TIME],
+                University.END_TIME: slot[University.END_TIME]
+            } for slot in time_slots]
+
+            course = {
+                University.COURSE_NAME: course_doc[University.COURSE_NAME],
+                University.PREFERRED_TIME_SLOTS: preferred_time_slots
+            }
+            preferred_courses.append(course)
+        except Exception as e:
+            return HttpResponseBadRequest(f"Error creating Course object for {course_name}: {str(e)}")
+
+    return preferred_courses
+
+
+def create_and_add_instructor(name, preferred_courses, availability):
+    instructor = {
+        University.KEY: name,
+        University.NAME: name,
+        University.PREFERRED_COURSES: preferred_courses,
+        University.AVAILABILITY: availability
+    }
+
+    if instructors_collection.has(instructor[University.KEY]):
+        return HttpResponseBadRequest('Instructor with this name already exists')
+    instructors_collection.insert(instructor)
+    return None
```

Comment:
As I wrote before. hepler functions should not return Http response. At most they can return boolean indicating the status